worker-document/src/main/java/com/github/cafdataprocessing/workers/document/impl/ApplicationImpl.java

```diff
@@ -182,4 +189,13 @@ private static String getFailureQueue(final DocumentWorkerConfiguration configur
             ? configuration.getOutputQueue()
             : failureQueue;
     }
+
+    private static String getInvalidQueue(final DocumentWorkerConfiguration configuration)
+    {
+        final String invalidQueue = configuration.getInvalidQueue();
+
+        return (invalidQueue == null || invalidQueue.isEmpty())
+            ? configuration.getOutputQueue()
```

Comment:
Shouldn't this be falling back to the failure queue?

```suggestion
            ? getFailureQueue(configuration)
```